testing/src/scenario/context.py

```diff
@@ -503,13 +507,18 @@ def __init__(
                 spec: _CharmSpec[CharmType] = _CharmSpec.autoload(charm_type)
             except MetadataNotFoundError as e:
                 raise ContextSetupError(
-                    f"Cannot setup scenario with `charm_type`={charm_type}. "
-                    f"Did you forget to pass `meta` to this Context?",
+                    f"Cannot automatically setup scenario with `charm_type`={charm_type}. "
+                    "This is likely because this charm has some nonstandard repository setup and Scenario "
+                    "can't find the charmcraft|(metadata|config|actions) yamls in their usual location. "
+                    "Please parse and provide their contents manually via the `meta`, `config` "
+                    "and `actions` Context parameters.",
                 ) from e
 
         else:
             if not meta:
-                meta = {"name": str(charm_type.__name__)}
+                raise ContextSetupError(
+                    "Scenario doesn't support overriding `actions|config` without overriding `meta` too."
+                )
```

Comment:
And I suppose here also :disappointed: 

```suggestion
                raise ContextSetupError(
                    "To override `actions|config` you must also override `meta`."
                )
```